index.js

```diff
@@ -629,6 +629,40 @@ const publicApi = {
         return this;
     },
 
+    /**
+     * If enabled, the eslint-loader is enabled.
+     *
+     * https://github.com/MoOx/eslint-loader
+     *
+     *     // enables the eslint loaded using the default eslint configuration.
+     *     Encore.enableEslintLoader();
+     *
+     *     // Optionally, you can pass in the configuration eslint should extend.
+     *     Encore.enableEslintLoader('airbnb');
+     *
+     *     // You can also pass in an object of options
+     *     // that will be passed on to the eslint-loader
+     *     Encore.enableEslintLoader({
+     *         extends: 'airbnb',
+               emitWarning: false
+     *     });
```

Comment:
Bah. So, this looks really great. But so far, we've avoided allowing users to pass in options like this... because of potential merging issues between their config and our config. I really can't think of an issue here - their config wins, should be pretty straightforward. But it breaks from our convention. @Lyrkan what do you think?

Reply:
Well, the main issue I have with it is that we don't do the same thing for other methods...
It isn't a really big issue since in this you can also use a callback but it could confuse people.

I'd say that we have to decide whether or not we'll implement the same kind of trick (and also always do `Object.assign(defaultConfig, userConfig)`) to other methods.

Reply:
Like lyrkan said, we should probably aggree on a common interface to configure the different plugins

Reply:
Ok, let's keep it. And then we can think about adding this on a case-by-case basis. I think sometimes it's *not* a good idea, because we'll need to merge the config. But in this case, even though we're merging the config, the config is flat and simple. There may be some cases where the merging could be more complex, and we could possibly not allow this then :)

Reply:
I can also remove this "nifty" merging and create a follow up ticket where we can further discuss/address. This to avoid breaking backward capabilities.

Reply:
As long as @Lyrkan agrees with the idea of starting to add this config where it makes sense, I'm cool with keeping it here for the new method. It won't break BC - it's just a "new philosophy" for us

Reply:
well, the callback API now support both editing the config to add more settings, and returning a new object. I'm not sure this automatic merging brings much value here